website/cue/reference/components/sources.cue

```diff
@@ -71,6 +71,127 @@ components: sources: [Name=string]: {
+									octet_counting:      "Byte frames according to the [octet counting](\(urls.rfc_6587_3_4_1)) format."
+								}
+								syntax: "literal"
+							}
+						}
+						character_delimited: {
+							description:   "Options for `character_delimited` framing."
+							required:      true
+							relevant_when: "method = `character_delimited`"
+							type: object: options: {
+								delimiter: {
+									description: "The character used to separate frames."
+									required:    true
+									type: string: {
+										examples: ["\n", "\t"]
+										syntax: "literal"
+									}
+								}
+								max_length: {
+									description: "The maximum frame length limit. Any frames longer than `max_length` bytes will be discarded entirely."
+									required:    false
+									common:      false
+									type: uint: {
+										default: 102400
+										examples: [65535, 102400]
+										unit: "bytes"
+									}
+								}
+							}
+						}
+						newline_delimited: {
+							description:   "Options for `newline_delimited` framing."
+							required:      false
+							common:        false
+							relevant_when: "method = `newline_delimited`"
+							type: object: options: {
+								max_length: {
+									description: "The maximum frame length limit. Any frames longer than `max_length` bytes will be discarded entirely."
+									required:    false
+									common:      false
+									type: uint: {
+										default: 102400
+										examples: [65535, 102400]
+										unit: "bytes"
+									}
+								}
+							}
+						}
+						octet_counting: {
+							description:   "Options for `octet_counting` framing."
+							required:      false
+							common:        false
+							relevant_when: "method = `octet_counting`"
+							type: object: options: {
+								max_length: {
+									description: "The maximum frame length limit. Any frames longer than `max_length` bytes will be discarded entirely."
+									required:    false
+									common:      false
+									type: uint: {
+										default: 102400
+										examples: [65535, 102400]
+										unit: "bytes"
+									}
+								}
+							}
+						}
+					}
+				}
+				decoding: {
+					common:      false
+					description: "Configures in which way frames are decoded into events."
+					required:    false
+					type: object: options: {
+						codec: {
+							description: "The decoding method."
+							required:    true
+							type: string: {
+								enum: {
+									bytes:  "Events containing the byte frame as-is."
+									json:   "Events being parsed from a JSON string."
+									syslog: "Events being parsed from a Syslog message."
+								}
+								syntax: "literal"
+							}
+						}
+						json: {
+							description:   "Options for `json` decoding."
+							required:      false
+							common:        false
+							relevant_when: "codec = `json`"
+							type: object: options: {
+								skip_empty: {
```

Comment:
Is this necessary to expose to the user? Why would anyone want to keep empty frames?

Reply:
One reason I can think of: If users want to receive errors/warnings when they ingest empty frames, in the same way as any other invalid JSON.

Reply:
It also says that in the description below:

> Whether empty frames are skipped without warning.

Would you prefer a more elaborate explanation there, or remove it all together?

Reply:
I've changed it to "not warn by default, but configurable" in 5016ac29fe416b3113f723460ab230cf7ef5f0ce. I think this is potentially useful information to the operator, just as if any other malformed strings would hit the parser.

Reply:
Sorry, went a bit trigger happy on the merge button. I see your point, though.

It is unlikely that anyone will ever complain that they did not find out they were sending lots of empty payloads to vector because the JSON parser did not report it.

Reply:
Followed up on it here: #9497.